api/v1alpha08/sonataflow_webhook.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package v1alpha08
+
+import (
+	"fmt"
+
+	cncfvalidator "github.com/serverlessworkflow/sdk-go/v2/validator"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/kiegroup/kogito-serverless-operator/api/metadata"
+	"github.com/kiegroup/kogito-serverless-operator/log"
+)
+
+func (r *SonataFlow) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// Uncomment to enable the mutating webhook.
+// Uncomment also the mutating webhook in the config/default/webhookcainjection_patch.yaml file.
+// //+kubebuilder:webhook:path=/mutate-sonataflow-org-v1alpha08-sonataflow,mutating=true,failurePolicy=fail,sideEffects=None,groups=sonataflow.org,resources=sonataflows,verbs=create;update,versions=v1alpha08,name=msonataflow.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &SonataFlow{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (s *SonataFlow) Default() {
+	klog.V(log.D).InfoS("Applying default values for ", "name", s.Name)
+	// Add defaults
+	if len(s.ObjectMeta.Annotations[metadata.Version]) == 0 {
+		s.ObjectMeta.Annotations[metadata.Version] = "0.0.1"
+	}
+	if len(s.ObjectMeta.Annotations[metadata.SpecVersion]) == 0 {
```

Comment:
We don't have SpecVersion in the annotations. It's the CRD version, see: https://github.com/kiegroup/kogito-serverless-operator/blob/main/api/v1alpha08/conversion.go#L134

Reply:
The `metadata.SpecVersion` is simply the supported spec version by the current operator's version.